### PR TITLE
site commands: status, delete and update

### PR DESCRIPTION
**Pull request by nluaces**
Implementation of the remaining site commands:
- skupper site create: allow to export the resource in yaml/json format
- skupper site update
- skupper site delete
- skupper site status

Because the commands are implementations suitable to kubernetes, they were moved to the kube package.

### CODE REVIEW

**Review by lynnemorrison — APPROVED** [review]

**Review by grs — APPROVED** [review]

File: internal/cmd/skupper/utils/encode.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+)
+
+func Encode(outputType string, resource interface{}) (string, error) {
+
+	var result []byte
+	var err error
+
+	switch outputType {
+	case "json":
+		result, err = json.Marshal(resource)
```
Comment (fgiorgetti) [inline internal/cmd/skupper/utils/encode.go:16]: We could use `MarshalIndent` here to have a pretty printed output, similar to the yaml version.

File: internal/cmd/skupper/site/kube/site_delete.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+	//Validate if there is already a site defined in the namespace
+	siteList, err := cmd.Client.Sites(cmd.Namespace).List(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		validationErrors = append(validationErrors, err)
+	} else if siteList == nil || (siteList != nil && len(siteList.Items) == 0) {
+		validationErrors = append(validationErrors, fmt.Errorf("there is no existing Skupper site resource to update"))
+	} else if len(siteList.Items) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("there are several sites in this namespace and and it should be only one"))
```
Comment (fgiorgetti) [inline internal/cmd/skupper/site/kube/site_delete.go:69]: If no site name is provided, then this beahivor makes sense, but if user enters the site name to
delete, then I understand it should delete the specified one

File: internal/cmd/skupper/site/kube/site_update.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+	resourceStringValidator := validator.NewResourceStringValidator()
+	linkAccessTypeValidator := validator.NewOptionValidator(utils.LinkAccessTypes)
+	outputTypeValidator := validator.NewOptionValidator(utils.OutputTypes)
+
+	//Validate if there is already a site defined in the namespace
+	siteList, _ := cmd.Client.Sites(cmd.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if siteList != nil && len(siteList.Items) == 0 {
+		validationErrors = append(validationErrors, fmt.Errorf("there is no existing Skupper site resource to update"))
+	} else if len(siteList.Items) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("there are several sites in this namespace and and it should be only one"))
```
Comment (fgiorgetti) [inline internal/cmd/skupper/site/kube/site_update.go:94]: same comment as for the site delete... if you specify the site name, then this should not be thrown

File: internal/cmd/skupper/root/root.go [hunk 4]
```diff
@@ -34,8 +37,9 @@ For more information visit https://skupper.io`,
 	rootCmd.AddCommand(link.NewCmdLink())
 	rootCmd.AddCommand(connector.NewCmdConnector())
 
-	//TODO: Add persistent flags for context and namespace
-	rootCmd.PersistentFlags().StringVarP(&config.Platform, "platform", "p", "", "The platform type to use [kubernetes, podman]")
+	rootCmd.PersistentFlags().StringVarP(&SelectedNamespace, "namespace", "n", "", "Set the namespace")
+	rootCmd.PersistentFlags().StringVarP(&SelectedContext, "context", "c", "", "Set the kubeconfig context")
```
[the remark this answers is not in the document]
Comment (nluaces) [reply internal/cmd/skupper/root/root.go:41]: It is defined like that in the docs... https://skupperproject.github.io/refdog/commands/site-create.html

[the remark this answers is not in the document]
Comment (nluaces) [reply internal/cmd/skupper/root/root.go:41]: @ssorj what do you think? should we keep `context` or use `kubeconfig`?

File: internal/cmd/skupper/root/root.go [hunk 5]
```diff
@@ -34,8 +37,9 @@ For more information visit https://skupper.io`,
...
+	rootCmd.PersistentFlags().StringVarP(&config.Platform, "platform", "p", "", "Set the platform type to use [kubernetes, podman]")
```
[the remark this answers is not in the document]
Comment (nluaces) [reply internal/cmd/skupper/root/root.go]: done! 17c770d23bb7501388e35c00df5d6d205c6254ad

File: internal/cmd/skupper/utils/encode.go (hunk 1 above)
Comment (nluaces) [reply internal/cmd/skupper/utils/encode.go]: done 17c770d23bb7501388e35c00df5d6d205c6254ad

File: internal/cmd/skupper/root/root.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (nluaces) [reply internal/cmd/skupper/root/root.go:41]: thanks for the clarification: I will add `kubeconfig` as well.

[the remark this answers is not in the document]
Comment (nluaces) [reply internal/cmd/skupper/root/root.go:42]: done dbc9d06417abaa819ee240dbc5abbaab1f6eb1b9

File: internal/cmd/skupper/site/kube/site_delete.go (hunk 2 above)
Comment (nluaces) [reply internal/cmd/skupper/site/kube/site_delete.go]: done! bd679654bdad99d6f2e64d2b0a4158fcdea3e02e

**Review by fgiorgetti — APPROVED** [review]

**Pull request merged**